providers/security/filesec/file_security.go

```diff
@@ -534,6 +539,54 @@ func (s *FileSecurity) VerifyCertificate(certpem []byte, name string) error {
 		return err
 	}
 
+	/*
+	* So, golang 1.15 removed support for certificates that effectively only have the CommonName
+	* in the subject.
+	*
+	* Unfortunately, by default puppet does not create certificates with DNS SAN records.
+	*
+	* This leaves us in a bit of a quandry - in order to validate the certificates for our usage,
+	* they must have SAN records, which contain the DNS record.
+	*
+	* This awful hack and workaround will attempt to detect if the SAN extension is present in the certificate,
+	* and if not will add one to the result in memory before we pass to the verify process.
+	 */
+
+	// This value is defined by Joint-ISO-ITU-T, I think.  This this safe to have hard coded
+	oidExtensionSubjectAltName := []int{2, 5, 29, 17}
+
+	// Default to the premise that the cert will have a SAN
+	hasSan := false
+
+	for _, e := range cert.Extensions {
+		if e.Id.Equal(oidExtensionSubjectAltName) {
+			hasSan = true
+		}
+	}
+
+	if !hasSan && s.conf.AppendSAN {
+		s.log.Debugf("Certificate for %s does not have a SAN, appending one", cert.Subject.CommonName)
+
+		// We do not have a SAN, so continue down this path of creating one
+		var e error
+		extSubjectAltName := pkix.Extension{Id: oidExtensionSubjectAltName, Critical: false}
+
+		extSubjectAltName.Value, e = asn1.Marshal([]string{"DNS:" + cert.Subject.CommonName})
+
+		// This value is usually populated when the certificate is parsed, this is necessary
+		// since the validation code in x509/verify.go will reference this array vs rebuilding
+		// the list of SANs every time
+		cert.DNSNames = append(cert.DNSNames, cert.Subject.CommonName)
+
+		if e != nil {
+			s.log.Error(e.Error())
```

Comment:
some more details here if we were going to log, though probably we can just skip logging all together as the callers will handle the error and log if they wish

Reply:
I don't know how it'd actually fail.  There's no validation on the marshalling as far as I can tell, unless some non-..ascii? data got in there.

Reply:
maybe :) or they might want to add validation in a future release and didnt want to then change the API?

Reply:
Right now, the only way that'd get in is if you were somehow able to marshal data to begin with, which would break the ASN1 interpreter.  

Or memory corruption.